### PR TITLE
Reader site stream spring clean

### PR DESCRIPTION
**Pull request by bluefuton**
Clean up `reader/site-stream` to follow current coding guidelines.

No functional changes.

### To test

Visit a site stream (e.g. http://calypso.localhost:3000/discover) and ensure that there are no display issues or errors in the console.

Visit an empty site stream (e.g.  http://calypso.localhost:3000/read/blogs/12345) and make sure the empty content message displays correctly:

screen shot 2016-11-17 at 13 09 16

### CODE REVIEW

**Comment by samouri** [comment]
Checking this out now:
 - I have to commend you on your commit-names & logical separation of changes! very organized and a joy to read through 👍

File: client/reader/site-stream/index.jsx [hunk 1]
```diff
@@ -27,55 +34,59 @@ function checkForRedirect( site ) {
...
 }
 
-const SiteStream = React.createClass( {
+class SiteStream extends React.Component {
 
-	getDefaultProps: function() {
-		return { showBack: true };
-	},
+	static propTypes = {
+		siteId: React.PropTypes.number.isRequired
```
Comment (samouri) [inline client/reader/site-stream/index.jsx]: In the empty stream I am getting:
```
vendor.development.js?v=da7a8da7db:3290 Warning: Failed prop type: Invalid prop `siteId` of type `string` supplied to `SiteStream`, expected `number`.
    in SiteStream (created by LocalizedSiteStream)
    in LocalizedSiteStream
    in Provider
```

This probably isn't a regression and was always there though so I don't mind landing this without fixing the propType issue

Comment (bluefuton) [inline client/reader/site-stream/index.jsx]: There were no prop types specified on the component before, so this is a new one. I'll fix!

**Comment by bluefuton** [comment]
Thanks for the 👀 @samouri!

**Pull request merged**